### PR TITLE
config for name engine

### PR DESCRIPTION
**Pull request by frostmark**

### CODE REVIEW

**Comment by accessd** [comment]
Hi! For what case engine name could be different?
I see that it's the defined helper: https://github.com/rails/rails/blob/5ccdd0bb6d1262a670645ddf3a9e334be4545dac/railties/lib/rails/application/finisher.rb#L51

**Comment by frostmark** [comment]
@accessd for example, when you connect into another engine

File: lib/rails-settings-ui.rb [hunk 1]
```diff
@@ -24,6 +24,9 @@ module RailsSettingsUi
   mattr_accessor :defaults_for_settings
   self.defaults_for_settings = {}
 
+  mattr_accessor :name_engine
+  self.name_engine = "main_app"
```
Comment (accessd) [inline lib/rails-settings-ui.rb]: let's call it `engine_name`

File: lib/rails-settings-ui/main_app_route_delegator.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ module RailsSettingsUi::MainAppRouteDelegator
   # delegate url helpers to main_app
```
Comment (accessd) [inline lib/rails-settings-ui/main_app_route_delegator.rb]: I think that we should rename `MainAppRouteDelegator` -> `RouteDelegator`

File: lib/rails-settings-ui/main_app_route_delegator.rb [hunk 3]
```diff
@@ -14,6 +14,6 @@ def respond_to?(method)
 
   private
   def main_app_route_method?(method)
```
Comment (accessd) [inline lib/rails-settings-ui/main_app_route_delegator.rb]: what about `main_app_route_method?` -> `engine_route_method?`

File: lib/rails-settings-ui/main_app_route_delegator.rb (hunk 2 above)
Comment (accessd) [inline lib/rails-settings-ui/main_app_route_delegator.rb]: `# delegate url helpers to another engine`

File: lib/rails-settings-ui.rb (hunk 1 above)
Comment (accessd) [reply lib/rails-settings-ui.rb]: and we need to add this option to README

**Comment by frostmark** [comment]
@accessd ok, i fix it

File: lib/rails-settings-ui.rb [hunk 4]
```diff
@@ -24,9 +24,12 @@ module RailsSettingsUi
   mattr_accessor :defaults_for_settings
   self.defaults_for_settings = {}
 
+  mattr_accessor :engine_name
+  self.engine_name = "main_app"
+
   class << self
-    def inline_main_app_routes!
-      ::RailsSettingsUi::ApplicationController.helper ::RailsSettingsUi::MainAppRouteDelegator
+    def inline_engine_routes!
```
Comment (accessd) [inline lib/rails-settings-ui.rb]: `inline_main_app_routes!` should stay with deprecation message: `warn "[DEPRECATION] `inline_main_app_routes!` is deprecated.  Please use `inline_engine_routes!` instead."`

File: lib/rails-settings-ui/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module RailsSettingsUi
-  VERSION = '1.3.1'
+  VERSION = '1.3.2'
```
Comment (accessd) [inline lib/rails-settings-ui/version.rb]: please add changes to `CHANGES.md` and change `inline_main_app_routes!` method in README + add description of the new config attribute `engine_name` to Config section of README

Comment (frostmark) [reply lib/rails-settings-ui/version.rb]: @accessd updated

File: lib/rails-settings-ui.rb [hunk 6]
```diff
@@ -24,9 +24,17 @@ module RailsSettingsUi
   mattr_accessor :defaults_for_settings
   self.defaults_for_settings = {}
 
+  mattr_accessor :engine_name
+  self.engine_name = "main_app"
+
   class << self
     def inline_main_app_routes!
-      ::RailsSettingsUi::ApplicationController.helper ::RailsSettingsUi::MainAppRouteDelegator
+      warn("[DEPRECATION] inline_main_app_routes!is deprecated. Please useinline_engine_routes!instead.")
```
Comment (accessd) [inline lib/rails-settings-ui.rb]: You missed spaces between words here

File: lib/rails-settings-ui.rb [hunk 7]
```diff
@@ -24,9 +24,17 @@ module RailsSettingsUi
...
+      ::RailsSettingsUi::ApplicationController.helper ::RailsSettingsUi::RouteDelegator
```
Comment (accessd) [inline lib/rails-settings-ui.rb]: just call `inline_engine_routes!`

File: lib/rails-settings-ui/version.rb (hunk 5 above)
Comment (accessd) [reply lib/rails-settings-ui/version.rb]: please add description of your contribution to CHANGES.md

Comment (frostmark) [reply lib/rails-settings-ui/version.rb]: @accessd i did :nerd_face:

**Pull request merged**